examples/bri-3/src/bri/workgroup/workflows/agents/workflowsStorage.agent.ts

```diff
@@ -0,0 +1,144 @@
+import { Injectable, NotFoundException } from '@nestjs/common';
+import { PrismaService } from '../../../../../prisma/prisma.service';
+import { Workstep } from '../../worksteps/models/workstep';
+import { WORKFLOW_NOT_FOUND_ERR_MESSAGE } from '../api/err.messages';
+import { Workflow } from '../models/workflow';
+
+@Injectable()
+export class WorkflowStorageAgent extends PrismaService {
+  async getWorkflowById(id: string): Promise<Workflow> {
+    const workflowModel = await this.workflow.findUnique({
+      where: { id: id },
+      include: { worksteps: true },
+    });
+
+    if (!workflowModel) {
+      throw new NotFoundException(WORKFLOW_NOT_FOUND_ERR_MESSAGE);
+    }
+
+    return new Workflow(
+      workflowModel.id,
+      workflowModel.name,
+      workflowModel.worksteps.map((w) => {
+        return new Workstep(
+          w.id,
+          w.name,
+          w.version,
+          w.status,
+          w.workgroupId,
+          w.securityPolicy,
+          w.privacyPolicy,
+        );
+      }),
+      workflowModel.workgroupId,
+    );
+  }
+
+  async getAllWorkflows(): Promise<Workflow[]> {
```

Comment:
again scope? @Kasshern

Reply:
@Therecanbeonlyone1969 any suggestions what to query on? Retrieve those belonging to specific workgroup? or a specific name? including certain worksteps. A combination of those?

Reply:
I am fine if we keep it as is for now. We 'll scope it as we move forward and get a better understanding of the needs of the BPI consumers.

Reply:
Will leave as is and scope the query when BPI consumer needs elucidated.